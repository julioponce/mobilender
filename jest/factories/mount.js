import {
  createLocalVue,
  mount as utilsMount,
} from '@vue/test-utils';
import CompositionApi from '@vue/composition-api';
import Router from 'vue-router';

const localVue = createLocalVue();

localVue.use(CompositionApi);
localVue.use(Router);

const div = document.createElement('div');
div.id = 'root';
document.body.appendChild(div);

const mount = ({
  component,
  vuetify = null,
  router = null,
  attachTo = '#root',
  propsData = {},
  slots = {},
  listeners = {},
  stubs = {},
  attrs = {},
  mocks = {},
}) => {
  return utilsMount(component, {
    localVue,
    vuetify,
    router,
    attachTo: attachTo,
    propsData: {
      ...propsData,
    },
    attrs: {
      ...attrs,
    },
    slots: {
      ...slots,
    },
    listeners: {
      ...listeners,
    },
    stubs: {
      ...stubs,
    },
    mocks: {
      ...mocks,
    },
  });
};

export default mount;
