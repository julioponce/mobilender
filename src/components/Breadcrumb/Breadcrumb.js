import { computed } from '@vue/composition-api';

export default {
  name: 'Breadcrumb',
  setup (props, context) {
    const route = computed(() => context.root.$route);

    const crumbs = computed(() => {
      const pathArray = route.value.path.split('/').filter((path) => !!path);

      const breadcrumbs = pathArray.reduce((breadcrumbArray, path, index) => {
        const currentRoute = route.value.matched[index + 1];
        breadcrumbArray.push({
          to: currentRoute.path,
          text: currentRoute.meta.breadCrumb || currentRoute.name,
        });

        return breadcrumbArray;
      }, []);

      return breadcrumbs;
    });

    return { crumbs };
  },
};
