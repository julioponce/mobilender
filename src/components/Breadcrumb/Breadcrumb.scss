.breadcrumb {
  margin-bottom: 10px;

  .v-breadcrumbs {
    padding: 1rem 0 0 0;

    &__divider {
      color: #ccc !important;
    }

    &__item {
      font-size: $font-size-sm;
      text-decoration: underline;
      text-transform: capitalize;
      color: white;

      &:hover,
      &:focus,
      &:active,
      &:visited {
        color: white;
      }

      &.router-link-active {
        color: #999 !important;
        text-decoration: none !important;
      }
    }
  }
}
