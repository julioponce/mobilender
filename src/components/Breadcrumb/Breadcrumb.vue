<template>
  <div class="breadcrumb">
    <v-breadcrumbs divider="/" :items="crumbs">
      <template v-slot:item="{ item }">
        <router-link
          class="v-breadcrumbs__item"
          :to="item.to"
          exact
        >
          {{ item.text }}
        </router-link>
      </template>
    </v-breadcrumbs>
  </div>
</template>

<script src='./Breadcrumb.js' />
<style lang='scss' src='./Breadcrumb.scss' />
