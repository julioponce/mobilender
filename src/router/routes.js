const Layout = () => import('@/components/Layout');
const Home = () => import('@/components/pages/Home');
const Login = () => import('@/components/pages/Login');
const PageContainer = () => import('@/components/PageContainer');
const CreditosGrupales = () => import('@/components/pages/CreditosGrupales');
const ControlCambios = () => import('@/components/pages/ControlCambios');
const Inspeccionar = () => import('@/components/pages/Inspeccionar');
const ListadoSoftware = () => import('@/components/pages/ListadoSoftware');
const Software = () => import('@/components/pages/Software');

const routes = [
  {
    path: '/',
    component: Layout,
    children: [
      {
        alias: '',
        path: 'login',
        name: 'Login',
        component: Login,
        meta: {},
      },
      {
        path: 'home',
        component: PageContainer,
        meta: { breadCrumb: 'Home' },
        children: [
          {
            path: '',
            name: 'Home',
            component: Home,
            meta: { privateAccess: true },
          },
          {
            path: 'buscar-creditos-grupales',
            component: PageContainer,
            meta: { breadCrumb: 'Buscar Creditos Grupales' },
            children: [
              {
                path: '',
                name: 'Buscar Creditos Grupales',
                component: CreditosGrupales,
                meta: { privateAccess: true },
              },
              {
                path: 'inspeccionar',
                component: PageContainer,
                meta: { breadCrumb: 'Inspeccionar' },
                children: [
                  {
                    path: '',
                    name: 'Inspeccionar',
                    component: Inspeccionar,
                    meta: { privateAccess: true },
                  },
                  {
                    path: 'control-de-cambios',
                    name: 'Control de Cambios',
                    component: ControlCambios,
                    meta: {
                      privateAccess: true,
                      breadCrumb: 'Control de Cambios',
                    },
                  },
                ],
              },
            ],
          },
          {
            path: 'listado-de-software',
            component: PageContainer,
            meta: { breadCrumb: 'Listado de Software' },
            children: [
              {
                path: '',
                name: 'Listado de Software',
                component: ListadoSoftware,
                meta: { privateAccess: true },
              },
              {
                path: ':id',
                component: Software,
                meta: { privateAccess: true },
              },
            ],
          },
        ],
      },
    ],
  },
];

export default routes;
