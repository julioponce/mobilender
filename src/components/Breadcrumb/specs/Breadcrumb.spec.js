import Vuetify from 'vuetify';
import Router from 'vue-router';
import mount from '../../../../jest/factories/mount';
import routes from '../../../router/routes';
import Breadcrumb from '../Breadcrumb.vue';

describe('Breadcrumb', () => {
  let vuetify;
  const router = new Router({
    routes: routes,
  });

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  test('Renders', () => {
    const wrapper = mount({
      component: Breadcrumb,
      router,
      vuetify,
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
