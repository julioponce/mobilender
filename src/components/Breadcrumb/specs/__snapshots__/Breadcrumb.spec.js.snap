// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb Renders 1`] = `
<div
  class="breadcrumb"
>
  <ul
    class="v-breadcrumbs theme--light"
  />
</div>
`;
